test/runtime/Policies.go

```diff
@@ -951,4 +952,135 @@ var _ = Describe("RuntimeValidatedPolicyImportTests", func() {
 		Expect(strings.TrimSpace(policyMapsInVM.GetStdOut())).To(Equal(expected), "Only %s PolicyMap should be present", expected)
 
 	})
+
+	It("checks policy trace output", func() {
+
+		httpd2Label := "id.httpd2"
+		httpd1Label := "id.httpd1"
+		allowedVerdict := "Final verdict: ALLOWED"
+
+		By("Checking policy trace by labels")
+
+		By(fmt.Sprintf("Importing policy that allows ingress to %s from the host and %s", httpd1Label, httpd2Label))
+
+		allowHttpd1IngressHostHttpd2 := fmt.Sprintf(`
+			[{
+    			"endpointSelector": {"matchLabels":{"id.httpd1":""}},
+    			"ingress": [{
+        			"fromEndpoints": [
+            			{"matchLabels":{"reserved:host":""}},
+            			{"matchLabels":{"id.httpd2":""}}
+					]
+    			}]
+			}]`)
+
+		err := helpers.RenderTemplateToFile(policyJSON, allowHttpd1IngressHostHttpd2, 0777)
+		defer os.Remove(policyJSON)
+		Expect(err).Should(BeNil())
+
+		path := helpers.GetFilePath(policyJSON)
+		_, err = vm.PolicyImportAndWait(path, helpers.HelperTimeout)
+		Expect(err).Should(BeNil(), "Error importing policy %s", path)
+
+		By(fmt.Sprintf("Verifying that trace says that %s can reach %s", httpd2Label, httpd1Label))
+
+		res := vm.Exec(fmt.Sprintf(`diff -Nru <(echo "%s") <(cilium policy trace -s %s -d %s | grep "Final verdict:")`, allowedVerdict, httpd2Label, httpd1Label))
```

Comment:
Here, why you're not using `Expect(res.Output().String()).Should(ContainSubstring("Final verdict: ALLOWED"))`? It's much easy no? and if you want to debug you'll get the full output in the log. 

This is already used in `k8st/Policies.go`

Reply:
Agreed, this is much better. Will have this done in next upload.

Reply:
Done.